TestModels/SimpleTypes/SimpleString/runtimes/rust/src/client.rs

```diff
@@ -0,0 +1,48 @@
+// Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+
+#[derive(Debug)]
+pub(crate) struct Handle {
+    pub(crate) conf: crate::Config,
+    pub(crate) inner: *mut dyn crate::implementation_from_dafny::r#_simple_dtypes_dsmithystring_dinternaldafny_dtypes::ISimpleTypesStringClient
+}
+
+#[derive(::std::clone::Clone, ::std::fmt::Debug)]
+pub struct Client {
+    handle: ::std::sync::Arc<Handle>,
+}
+
+impl Client {
+    /// Creates a new client from the service [`Config`](crate::Config).
+    #[track_caller]
+    pub fn from_conf(conf: crate::Config) -> Self {
```

Comment:
I think this will ave to be
-> Result<Self, Error>

Reply:
I think so. Interestingly the `smithy-rs` generated version has:

```rust
if let Err(err) = Self::validate_config(&handle) {
    panic!("Invalid client configuration: {err}");
}
```

So invalid config just causes a panic. I expect we at least need to allow constructing SDK wrappers to fail since we could hit arbitrary target language errors? Do we also need this for local services?

Reply:
We should never panic! -- only return Results.
For a library "panic!" means "we've lost internal consistency and cannot guarantee the state of your program".

It's the customer's code that should
`if let Err(err) = Self::validate_config(&handle) {`
our libraries should only propagate errors.

Reply:
Consulted with the author of `smithy-rs` on this. It turns out to be a UX improvement, because failing to construct a client is through to be generally unrecoverable, and not something you would retry or fallback to an alternate configuration. In other words, if the only reasonable thing for callers to do on an `Err` is panic, the client might as well do it for you. 

They did say it's totally reasonable to go the other way, so I'll change to use a Result instead.